spacialgaze-plugins/news.js

```diff
@@ -1,40 +1,34 @@
-/*
-* News System for SpacialGaze
-* Credits: Lord Haji, HoeenHero
-*/
+/**
+ * News System for SpacialGaze
+ * This Shows News via the /news view command and sends news ns PMs when users connect to the server if they have subscribed
+ * Uses nef to add News to nef's json database
+ * Credits: Lord Haji, HoeenHero
+ * @license MIT license
+ */
 
 'use strict';
 
-function generateNews(user) {
+function generateNews(userid) {
 	let newsData, newsDisplay = [];
-	user = toId(user);
-	Db.news.keys().forEach(announcement => {
-		newsData = Db.news.get(announcement);
-		newsDisplay.push(`<h4>${announcement}</h4>${newsData[1]}<br /><br />—${SG.nameColor(newsData[0], true)} <small>on ${newsData[2]}</small>`);
-	});
+	let keys = Db.news.keys();
+	for (let i = 0; i < keys.length; i++) {
+		newsData = Db.news.get(keys[i]);
+		newsDisplay.push(`<h4>${keys[i]}</h4>${newsData[1]}<br /><br />—${SG.nameColor(newsData[0], true)} <small>on ${newsData[2]}</small>`);
+	}
 	return newsDisplay;
 }
 
-function hasSubscribed(user) {
-	if (typeof user === 'object') user = user.userid;
-	if (Db.NewsSubscribers.has(toId(user))) return true;
-	return false;
-}
-
-function showSubButton(user) {
-	user = toId(user);
-	let output;
-	output = "<hr><center><button class = \"button\" name=\"send\" value=\"/news " + (hasSubscribed(user) ? "unsubscribe" : "subscribe") + "\">" + (hasSubscribed(user) ? "Unsubscribe from the news" : "Subscribe to the news") + "</button></center>";
-	return output;
+function showSubButton(userid) {
+	let hasSubscribed = Db.NewsSubscribers.get(userid, false);
+	let subStatus = [(hasSubscribed ? `unsubscribe` : `subscribe`), (hasSubscribed ? `Unsubscribe from the news` : `Subscribe to the news`)];
```

Comment:
you could just put the conditional ternary operators into the the next line instead of declaring an extra array.